utils/src/main/scala/za/co/absa/enceladus/utils/general/Algorithms.scala

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.utils.general
+
+import scala.collection.mutable.ListBuffer
+
+object Algorithms {
+
+  /**
+    * A stable group by takes a collection of items and groups items when consecutive items have the same
+    * grouping value. A grouping value is provided by a callback function.
+    *
+    * The function does not change the original order of the elements.
+    *
+    * @param collection     An input collection
+    * @param groupedElement A callback that for a given item produces a value by which items should be grouped
+    * @return A sequence of groups of the original items
+    */
+  def stableGroupBy[T, A](collection: Seq[T], groupedElement: T => A): Seq[Seq[T]] = {
```

Comment:
```suggestion
  def stableGroupBy[T, A](collection: Seq[T], groupBy: T => A): Seq[Seq[T]] = {
```
The second parameter is a callback function, I'd give it a more "verb-y" name  to reflect that. (The suggestion above is not the best option, I'm sure you can come up with something more appropriate)

Reply:
In Scala standard library this callback function is named `f`. We can follow them...